cmd/thanos/query.go

```diff
@@ -96,6 +96,9 @@ func registerQuery(app *extkingpin.App) {
 	selectorLabels := cmd.Flag("selector-label", "Query selector labels that will be exposed in info endpoint (repeated).").
 		PlaceHolder("<name>=\"<value>\"").Strings()
 
+	endpoints := cmd.Flag("endpoint", "Addresses of statically configured Thanos API servers (repeatable). The scheme may be prefixed with 'dns+' or 'dnssrv+' to detect Thanos API servers through respective DNS lookups.").
```

Comment:
I would suggest removing `store` at this point. Or are we planning that for future PRs?

Reply:
Essentially what's the deprecation plan? Can we note the deprecation plan in the store flag comment and changelog?

Reply:
If we only need to keep one of `store` and `endpoint`, what about just reusing `store`?

Reply:
According to [proposal](https://thanos.io/tip/proposals-accepted/202101-endpoint-discovery.md/), we were planning to remove old flags after some grace period (maybe after 1-2 releases).

Reply:
OK

Reply:
Can we say about `deprecation` for store flag then?

Reply:
Yes sure